hoomd/hpmc/external/field.py

```diff
@@ -1,243 +1,223 @@
+            elif isinstance(integrator, integrate.ConvexSpheropolygon):
+                cls = _hpmc.ExternalFieldHarmonicSpheropolygon
+            elif isinstance(integrator, integrate.FacetedEllipsoid):
+                cls = _hpmc.ExternalFieldHarmonicFacetedEllipsoid
+            elif isinstance(integrator, integrate.Polyhedron):
+                cls = _hpmc.ExternalFieldHarmonicPolyhedron
+            elif isinstance(integrator, integrate.Sphinx):
+                cls = _hpmc.ExternalFieldHarmonicSphinx
+            elif isinstance(integrator, integrate.SphereUnion):
+                cls = _hpmc.ExternalFieldHarmonicSphereUnion
+            elif isinstance(integrator, integrate.FacetedEllipsoidUnion):
+                cls = _hpmc.ExternalFieldHarmonicFacetedEllipsoidUnion
+            elif isinstance(integrator, integrate.ConvexSpheropolyhedronUnion):
+                cls = _hpmc.ExternalFieldHarmonicConvexPolyhedronUnion
+            else:
+                msg = 'Error initializing hoomd.hpmc.field.Harmonic: '
+                msg += 'unsupported integrator'
+                raise RuntimeError(msg)
+        else:
+            msg = 'Error initializing hoomd.hpmc.field.Harmonic: '
+            msg += 'GPU not supported.'
+            raise RuntimeError(msg)
+
+        self._cpp_obj = cls(
+            cpp_sys_def,
+            self.reference_positions,
+            self.k_translational,
+            self.reference_orientations,
+            self.k_rotational,
+            self.symmetries,
+        )
+        super()._attach()
+
+    @log(requires_run=True)
+    def energy(self):
+        """float: The energy of the harmonic field :math:`[\\mathrm{energy}]`.  # noqa
 
-    def get_energy(self):
-        R"""    Get the current energy of the lattice field.
-                This is a collective call and must be called on all ranks.
-        Example::
-            mc = hpmc.integrate.sphere(seed=415236);
-            lattice = hpmc.field.lattice_field(mc=mc, position=fcc_lattice, k=1000.0);
-            run(20000)
-            eng = lattice.get_energy()
         """
-        timestep = hoomd.context.current.system.getCurrentTimeStep()
-        return self.cpp_compute.getEnergy(timestep)
+        timestep = self._simulation.timestep
+        return self._cpp_obj.getEnergy(timestep, True, True)
 
-    def get_average_energy(self):
-        R"""    Get the average energy per particle of the lattice field.
-                This is a collective call and must be called on all ranks.
-
-        Example::
-            mc = hpmc.integrate.sphere(seed=415236);
-            lattice = hpmc.field.lattice_field(mc=mc, position=fcc_lattice, k=exp(15));
-            run(20000)
-            avg_eng = lattice.get_average_energy() //  should be about 1.5kT
+    @log(requires_run=True)
+    def energy_translational(self):
+        """float: The energy associated with positional fluctuations \
+            :math:`[\\mathrm{energy}]`.  # noqa
```

Comment:
```suggestion
            :math:`[\\mathrm{energy}]`. 
```
Don't include noqa in docstrings.